File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/PlayerInput.java
package net.mjduffin.risk.lib.usecase;

import java.util.Map;

public interface PlayerInput {


    void draft(String playerName, Map<String, Integer> draft) throws GameplayException;
    void draftSingle(String playerName, String territory, int units) throws GameplayException;
    AttackResult attack(String playerName, String attackingTerritory, String defendingTerritory) throws PlayerNotFoundException, TerritoryNotFoundException, GameplayException;
    void endAttack(String playerName) throws PlayerNotFoundException;
    void move(String playerNmame, int units) throws PlayerNotFoundException;
    void fortify(String playerName, String fromTerritory, String toTerritory, int units) throws PlayerNotFoundException, TerritoryNotFoundException, GameplayException;

}

File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/PlayerInput.kt
package net.mjduffin.risk.lib.usecase

interface PlayerInput {
    fun draft(playerName: String, draft: Map<String, Int>)

    fun draftSingle(playerName: String, territory: String, units: Int)

    fun attack(playerName: String, attackingTerritory: String, defendingTerritory: String): AttackResult

    fun endAttack(playerName: String)

    fun move(playerName: String, units: Int)

    fun fortify(playerName: String, fromTerritory: String, toTerritory: String, units: Int)
}